### PR TITLE
Allow translation of language name into other languages

### PR DESCRIPTION
**Pull request by phil-davis**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #5110 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Add some translations of language names - French for French is "Francais", for German is "Allemande" etc.
2. Login and got to the account settings page.
3. Change you language to French, apply and refresh.
4. The selected language should be "Francais"
5. Look in the language dropdown, see the "Allemande" (German) is there, choose it.
6. Apply and refresh.
7. The selected language says "Deutsche".
8. Look in the dropdown for "Türkisch" - it should be there.

#### List backwards compatibility breaks:
1. Perhaps this should be a system option - because some installs currently with the system language set to other than English, specially those that use another script (Hindi, East Asian languages...) may or may not want the dropdown list to be localized. e.g. for a non-Chinese person to choose their language from a dropdown list that is all in Chinese script is "difficult". And the same the other way, a KOrean person who does not know latin script struggles to select a language.
Note: "English - United States" seems to "stick" in the list. So that is some sort of common reference point. If I arrive on a Chinese installation I can first choose "English - United States" then pick the language I really want from the dropdown list in English.

### CODE REVIEW

File: app/bundles/CoreBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -228,6 +228,12 @@ mautic.core.ip_lookup.last_updated="Last updated on %date%"
 mautic.core.ipaddress="IP address"
 mautic.core.label="Label"
 mautic.core.language="Language"
+mautic.core.language.en="English"
+mautic.core.language.en_GB="English (Pommy)"
+mautic.core.language.en_US="English (USA)"
+mautic.core.language.fr="French"
+mautic.core.language.de="German"
+mautic.core.language.tr="Turkish"
```
Comment (phil-davis) [inline app/bundles/CoreBundle/Translations/en_US/messages.ini]: This is an example list, which can be filled out if this seems a "good thing". The translators that are making the ``fr/message.ini`` ``de/messages.ini`` and so on can then put these keys with strings that are the names of the languages from their point of view "Francais", "Allemand", "Deutsche" etc.

Maybe there is somewhere that already has a matrix lookup table of every known language name from the point of view of every other language?

**Comment by phil-davis** [comment]
Rebased

**Comment by phil-davis** [comment]
Rebased on top of #5367 so that CI should pass.

**Comment by phil-davis** [comment]
Rebased again

**Comment by tedcallier** [comment]
I am all for having language names in all languages. But isn't this PR a solution to something that isn't a problem?

Who hasn't ever set the phone language of a friend to Taiwanese, or Japanese, waiting to see his face when desperately trying to figure our how to set the language back to something he understands?

It would be more helpful if all language names would be shown in their native language. A native French, hooked into an English user-interface doesn't necessarily realise that "Deutsch" is the translation of "Allemand" while he is seeing "German".

**Comment by phil-davis** [comment]
I agree, it is tricky - actually it is probably better (in all software with language selection) to show 2 things for every language in the drop-down list:
- the name of the language as written by the current language
- the name of the language as written by its native speakers

So when the current language is English, there would be entries like:
```
German - Deutsch
Nepalese - नेपाली
```

I did this PR when I was switching around languages in the UI just testing that stuff seemed to work OK. It seemed easy enough to add in. But certainly not a big issue!

I like to have "as full as possible/reasonable" multi-lingual support in software.

**Comment by alanhartless** [comment]
I'm fine with this if you want to finish up adding the translation strings for the supported languages.

**Comment by dbhurley** [comment]
@phil-davis - update?

**Comment by alanhartless** [comment]
Closing this as it's very old. If you'd like it re-considered, please submit an updated PR.

**Pull request closed without merging**